Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MptrXmlElementAccess.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.dataformat.access;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+import javax.xml.bind.DataBindingException;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.kitodo.api.dataformat.LinkedStructure;
+import org.kitodo.api.dataformat.Structure;
+import org.kitodo.dataformat.metskitodo.DivType;
+import org.kitodo.dataformat.metskitodo.Mets;
+
+public class MptrXmlElementAccess {
+
+    LinkedStructure linkedStructure = new LinkedStructure();
+
+    /**
+     * Constructor to read a linked structure from METS.
+     * 
+     * @param div
+     *            METS {@code <div>} element from which the structure is to be
+     *            built
+     * @param mets
+     *            METS data structure of the current workpiece
+     * @param getInputStreamFunction
+     *            A reference to a function
+     *            {@code InputStream getInputStream(URI uri, Boolean couldHaveToBeWrittenInTheFuture)}.
+     *            If invoked, the calling function is responsible of closing the
+     *            stream.
+     * @throws IllegalStateException
+     *             if the child does not have a link to the parent, or the
+     *             parent link of the child returns METS data different from the
+     *             parent’s METS
+     */
+    MptrXmlElementAccess(DivType div, Mets parent, Function<Pair<URI, Boolean>, InputStream> getInputStreamFunction) {
+        try {
+            linkedStructure.setOrder(div.getORDER());
+            URI uri = new URI(div.getMptr().stream().findFirst().get().getHref());
+            linkedStructure.setUri(uri);
+            Mets child;
+            try (InputStream in = getInputStreamFunction.apply(Pair.of(uri, true))) {
+                child = MetsXmlElementAccess.readMets(in);
+            }
+            ensureParenthood(parent, child, getInputStreamFunction);
+            Structure linked = MetsXmlElementAccess.toWorkpiece(child, getInputStreamFunction).getStructure();
+            linkedStructure.setLabel(linked.getLabel());
+            linkedStructure.setType(linked.getType());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (URISyntaxException e) {
+            throw new DataBindingException(e.getMessage(), e);
+        }
```

Comment:
please add error messages

Reply:
I do not understand your request, there is an error message. What did you mean?